sous-chef/sous_chef/recipe_book/read_recipe_book.py

```diff
@@ -157,6 +210,33 @@ def _check_total_time(recipe: pd.Series):
         if recipe.time_total is pd.NaT:
             raise RecipeTotalTimeUndefinedError(recipe_title=recipe.title)
 
+    def _construct_filter(self, row: pd.Series, filter_str: str):
+        def _replace_entity(entity_name: str, match_obj: re.Match) -> str:
+            if (entity := match_obj.group(1)) is not None:
+                if (
+                    entity.lower()
+                    not in {
+                        "tag": self.tag_tuple,
+                        "category": self.category_tuple,
+                    }[entity_name]
+                ):
+                    raise RecipeLabelNotFoundError(
+                        field=entity_name, search_term=entity
+                    )
+                return (
+                    f"('{entity.lower()}' in "
+                    f"{row.tags if entity_name == 'tag' else row.categories})"
+                )
+
+        # TODO return boolean numpy array from eval & df with iterate over row
+        for s, l in zip(["c", "t"], ["category", "tag"]):
```

Comment:
Please give variables in for loop descriptive names, instead of s, l

Reply:
improved hopefully